sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs

```diff
@@ -537,31 +535,99 @@ private string ComputeNamespace(DirectoryBaseViewModel directory)
             }
         }
 
-        private async Task<List<AssetViewModel>> InvokeAddAssetTemplate(LoggerResult logger, string name, DirectoryBaseViewModel directory, TemplateAssetDescription templateDescription, IEnumerable<UFile> files, PropertyContainer? customParameters)
+        private async Task<string> GetAssetCopyDirectory(DirectoryBaseViewModel directory, UFile file)
         {
-            List<AssetViewModel> newAssets = new List<AssetViewModel>();
-            if (files is not null)
-            {
-                foreach (var file in files)
+            string path = directory.Path;
+            var message = Tr._p("Message", "Do you want to place the resource in the default location ?").ToFormat(file.FullPath);
+            string finalPath = directory.Package.Package.ResourceFolders[0] + "/" + path + file.GetFileName();
+            var pathResult = await Dialogs.MessageBox(message, MessageBoxButton.YesNo, MessageBoxImage.Question);
+            if (pathResult == MessageBoxResult.No)
+            {
+                bool done = false;
+                do
                 {
-                    bool inResourceFolder = false;
-                    foreach (var resourceFolder in directory.Package.Package.ResourceFolders)
+                    var fileDialog = Dialogs.CreateFileSaveModalDialog();
+                    fileDialog.Filters = new List<FileDialogFilter>() { new FileDialogFilter("", file.GetFileExtension()) };
+                    fileDialog.InitialDirectory = directory.Package.Package.ResourceFolders.FirstOrDefault().FullPath.Replace("/", "\\");
+                    fileDialog.DefaultFileName = file.GetFileName();
+                    DialogResult result = await fileDialog.ShowModal();
+
+                    // If the user closes the dialog, assume that they want to use the default directory
+                    if (result != DialogResult.Ok)
+                    {
+                        done = true;
+                    }
+                    else
                     {
-                        if (file.FullPath.StartsWith(resourceFolder.FullPath))
+                        var fullPath = Path.GetFullPath(fileDialog.FilePath).Replace("\\", "/");
```

Comment:
Should skip the replace under non-windows, you can test against `Path.DirectorySeparatorChar` or os directly. The output path should already be using forward slash in those OS for dir separator.
Without that check, if the directory name or file name contains a backslash in those platforms it'll break.